File: LeetCode/May challenge/Day 28/Day_28.py
class Solution:
    def countBits(self, num: int) -> List[int]:
        i, b = 0, 1
        bits = [0] * (num + 1)
        while b <= num:
            while i < b and i + b <= num:
                bits[i + b] = bits[i] + 1
                i += 1
            i = 0; b = b * 2
        return bits

File: LeetCode/May challenge/Day 28/Day_28.cpp
class Solution {
public:
    vector<int> countBits(int num) {
        vector<int> res;
        for(int i=0;i<=num;i++)
        {
            bitset<64> temp(i);// bitset converts the int to its binary 
            res.push_back(temp.count());//count in bitset ciunts the no. of set bits
        }
        return res;
    }
};
